Bottom: 09bb478a0c76a6cafb7b7e882519a191d5c452ab
Top:    cb7ae1732ca0d00c18b439869a6acc8d98fc6d1a
Author: Ben Greear <greearb@candelatech.com>
Date:   2015-09-22 17:14:05 -0700

mac80211: Add some debugging messages for set-bitrates logic.

Helps someone understand why things are broken.

Signed-off-by: Ben Greear <greearb@candelatech.com>


---

diff --git a/net/mac80211/cfg.c b/net/mac80211/cfg.c
index 8acfb99..691f987 100644
--- a/net/mac80211/cfg.c
+++ b/net/mac80211/cfg.c
@@ -2510,8 +2510,11 @@ static int ieee80211_set_bitrate_mask(struct wiphy *wiphy,
 
 	if (ieee80211_hw_check(&local->hw, HAS_RATE_CONTROL)) {
 		ret = drv_set_bitrate_mask(local, sdata, mask);
-		if (ret)
+		if (ret) {
+			pr_err("%s: drv-set-bitrate-mask had error return: %d\n",
+			       sdata->dev->name, ret);
 			return ret;
+		}
 	}
 
 	for (i = 0; i < NUM_NL80211_BANDS; i++) {
diff --git a/net/mac80211/driver-ops.h b/net/mac80211/driver-ops.h
index 1850a78..a5930f4 100644
--- a/net/mac80211/driver-ops.h
+++ b/net/mac80211/driver-ops.h
@@ -776,8 +776,10 @@ static inline int drv_set_bitrate_mask(struct ieee80211_local *local,
 
 	might_sleep();
 
-	if (!check_sdata_in_driver(sdata))
+	if (!check_sdata_in_driver(sdata)) {
+		pr_err("drv-set-bitrate-mask, sdata is not in driver!\n");
 		return -EIO;
+	}
 
 	trace_drv_set_bitrate_mask(local, sdata, mask);
 	if (local->ops->set_bitrate_mask)
diff --git a/net/mac80211/mlme.c b/net/mac80211/mlme.c
index 933dbb8..aa4284d 100644
--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -4729,7 +4729,7 @@ int ieee80211_mgd_assoc(struct ieee80211_sub_if_data *sdata,
 		int q = 0;
 
 		/* Skip CCK rates for 5Ghz band */
-		if (band == IEEE80211_BAND_5GHZ)
+		if (band == NL80211_BAND_5GHZ)
 			msk = msk << 4;
 
 #if 0
